pkg/kubelet/status/status_manager.go

```diff
@@ -381,6 +383,10 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 		klog.Errorf("Status update on pod %v/%v aborted: %v", pod.Namespace, pod.Name, err)
 		return false
 	}
+	if err := checkContainerStateTransition(oldStatus.EphemeralContainerStatuses, status.EphemeralContainerStatuses, pod.Spec.RestartPolicy); err != nil {
```

Comment:
can you wrap this in a feature gate check?

every time i hear about a new type of container proposed, i would love to point at this pr for how bugs can be found.  it would be good to write-up something that says "so you want to add a new type of container to pods, make sure you look in these areas"

is that something you want to help with @tedyu ?  its the type of thing side car containers would hit as an example.

Reply:
@derekwaynecarr 
I have added feature gate check.

I am willing to help with introduction of new container type(s).